website/src/components/Demo.tsx

```diff
@@ -49,16 +49,36 @@ function useCounterContract() {
 }
 
 function DemoAccount() {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
```

Comment:
this is an example of how it would look to interact with connectors with preauth. since it is async the logic is more complex. we should probably internalize some of this logic, similar to what wagmi does. perhaps we want to still support the advanced use case though?

im happy to propose the abstraction as part of this PR or as a follow up

Reply:
yup! moved it in and it's much cleaner now